browser/src/core/Socket.js

```diff
@@ -459,7 +484,14 @@ app.definitions.Socket = L.Class.extend({
 	// process so - slurp and then emit at idle - its faster to delay!
 	_slurpMessage: function(e) {
 		this._extractTextImg(e);
+		if (e.image && e.image.rawData && this._worker) {
+			this._queueWorkerTileProcessing(e);
```

Comment:
Am I reading this right that we take 'tile:' messages only - and we potentially then emit them out of order with subsequent 'delta:' and 'update:' messages which could arrive before the tiles are de-compressed by the thread - that is likely to lead to corrupted pixels.
Hmm.